tl/features/cell_context_matches.py

```diff
@@ -213,16 +218,97 @@ def initialize(self, input_df, context_dict, label_column):
+            context_similarity_list.append(similarity_matched)
+            context_property_list.append(property_matched)
+        # input_df['context_property'] = context_property_list
+        # input_df['context_similarity'] = context_similarity_list
+        input_df['context_score'] = context_score_list
+
+    def compute_property_scores(self, row_col_label_dict, n_cntxt_columns, num_rows):
+        # To calculate property score
+        properties_df = pd.DataFrame()
+        for r_c in row_col_label_dict:
+            row_col = r_c.split("_")
+            row_1 = row_col[0]
+            col_1 = row_col[1]
+            for cols in range(n_cntxt_columns):
+                if int(cols) != int(col_1):
+                    m = self.ccm_dict[r_c].get_properties(cols, q_node_dependent=True)
+                    int_prop = pd.DataFrame(m, columns = ["property_", "type", "best_score", "n_occurences"])
+                    int_prop['row'] = int(row_1)
+                    int_prop['column'] = int(col_1)
+                    int_prop['col_2'] = int(cols)
+                    int_prop['inv_occ'] = 1 / (int_prop['n_occurences'])
+                    properties_df = pd.concat([properties_df, int_prop])
+        property_value_list= []
+        grouped_obj = properties_df.groupby(['column','col_2', 'property_'])
+        for cell, group in grouped_obj:
+            property_score = (group['inv_occ'].sum(axis = 0))/num_rows
+            property_value_list.append([cell[2], cell[0], cell[1], property_score])
+        #
+        property_value_df = pd.DataFrame(property_value_list, columns = ['property_', 'column', 'col_2', 'property_score'])
+        property_value_df = property_value_df.sort_values(by=['column', 'property_score'], ascending=[True, False])
+        most_important_property_dict = property_value_df.drop_duplicates(['column', 'col_2'], keep = 'first')
+        return property_value_df, most_important_property_dict
+
     def compute_context_similarity(self,
                                    kg_id_context: List[dict],
                                    col2_string: str) -> List[dict]:
         result = []
 
         if col2_string is None:
             return result
+        # Determining the type of col_2 string
+        new_v = col2_string.replace('"', '')
+        to_match_1 = new_v.replace(",", "")
+        to_match_2 = to_match_1.replace(".", "0")
+        num_v = None
+        if " " in to_match_2:
+            split_v = to_match_1.split(" ")
+            for s in split_v:
+                if not s == ".":
+                    new_s = s.replace(".", "0")
+                    if new_s.isnumeric():
+                        num_v = s
+
+        if (to_match_1.isnumeric() or to_match_2.isnumeric() or num_v is not None) and (to_match_1.count(".") <= 1):
+            col2_string_datatype = ["d", "q"]
+        else:
+            col2_string_datatype = ["i"]
+
+        if "i" in col2_string_datatype and "," in col2_string_datatype:
```

Comment:
what is going on here? t

Reply:
This is for separators. Each token split by the separator will be individually tried to match.